HighwayPatrol.cpp

```diff
@@ -28,10 +33,25 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
-        assert(false);
+        if(Car* car = dynamic_cast<Car*>(v))
+        {
+            vehicleType = "Car";
+        }
+        else if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v))
+        {
+            vehicleType = "MotorCycle";
+        }
+        else if (SemiTruck* truck = dynamic_cast<SemiTruck*>(v))
+        {
+            vehicleType = "SemiTruck";
+        }
+       
         //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        std::cout << name << ": YOU IN THE [ " <<vehicleType<< " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
```

Comment:
limit the scope of `vehicleType` to just this `if( willArrest )` block
That is the only place it is used.  it does not need to exist outside of this scope.
- [ ] revise